source/adaptive_load/scoring_function_impl.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include "envoy/registry/registry.h"
+
+#include "nighthawk/adaptive_load/scoring_function.h"
+
+#include "api/adaptive_load/scoring_function_impl.pb.h"
+
+namespace Nighthawk {
+
+/**
+ * ScoringFunction that returns 1.0 when a metric is within thresholds and -1.0 otherwise.
+ * Supports upper or lower threshold or both.
+ */
+class BinaryScoringFunction : public ScoringFunction {
+public:
+  explicit BinaryScoringFunction(
```

Comment:
Let's document the constructors.

Here and below.

Reply:
Done.